dvc/external_repo.py

```diff
@@ -68,6 +68,13 @@ def _external_repo(url=None, rev=None, cache_dir=None):
     if rev is not None:
         _git_checkout(new_path, rev)
 
+    # If the external repo does not have a '.dvc' directory (i.e. it is a
+    # plain git repo without DVC metadata) then create one here
+    new_path_dvc_dir = os.path.join(new_path, Repo.DVC_DIR)
+    if not os.path.isdir(new_path_dvc_dir):
+        os.mkdir(new_path_dvc_dir)
+        Config.init(new_path_dvc_dir)
+
     repo = Repo(new_path)
```

Comment:
We've discussed it in the issue earlier and ideally, we shouldn't go this hacky route unless there are some really good reasons. Are there any? 🙂

Reply:
I thought of a better way of doing this now.